common/remote/hooks/azure/azurefile.cpp

```diff
@@ -779,6 +786,279 @@ static IFile *createAzureFile(const char *azureFileName)
+                status = ApiCopyStatus::Success;
+            else if (tstatus==Shares::Models::CopyStatus::Pending)
+                status = ApiCopyStatus::Pending;
+            else if (tstatus==Shares::Models::CopyStatus::Aborted)
+                status = ApiCopyStatus::Aborted;
+            else
+                status = ApiCopyStatus::Failed;
+        }
+        catch(const Azure::Core::RequestFailedException& e)
+        {
+            IERRLOG("Transfer using API .Poll() failed: %s (code %d)", e.ReasonPhrase.c_str(), static_cast<int>(e.StatusCode));
+            status = ApiCopyStatus::Failed;
+        }
+        return status;
+    }
+    virtual ApiCopyStatus abortCopy() override
+    {
+        if (status==ApiCopyStatus::Aborted || status==ApiCopyStatus::Failed)
+            return status;
+        int64_t outputLength;
+        CDateTime dateTime;
+        status = getProgress(dateTime, outputLength);
+        switch (status)
+        {
+            case ApiCopyStatus::Pending:
+                try
+                {
+                    if (fileCopyOp.HasValue() && fileCopyOp.Value().CopyId.HasValue())
+                        fileClient->AbortCopy(fileCopyOp.Value().CopyId.Value().c_str());
+                    else
+                        IERRLOG("AzureFileClient::AbortCopy() failed: CopyId is empty");
+                    status = ApiCopyStatus::Aborted;
+                }
+                catch(const Azure::Core::RequestFailedException& e)
+                {
+                    IERRLOG("Abort copy operation failed: %s (code %d)", e.ReasonPhrase.c_str(), static_cast<int>(e.StatusCode));
+                    status = ApiCopyStatus::Failed;
+                }
+                // fallthrough to delete any file remnants
+            case ApiCopyStatus::Success:
+                // already copied-> need to delete
+                try
+                {
+                    fileClient->Delete();
+                    status = ApiCopyStatus::Aborted;
+                }
+                catch(const Azure::Core::RequestFailedException& e)
+                {
+                    IERRLOG("Delete operation failed: %s (code %d)", e.ReasonPhrase.c_str(), static_cast<int>(e.StatusCode));
+                    status = ApiCopyStatus::Failed;
+                }
+                break;
+        }
+        return status;
+    }
+    virtual ApiCopyStatus getStatus() const override
+    {
+        return status;
+    }
+private:
+    ApiCopyStatus status = ApiCopyStatus::NotStarted;
+};
+
+class AzureBlobClient : public CInterfaceOf<IAPICopyClientOp>
+{
+    std::unique_ptr<BlobClient> blobClient;
+    StartBlobCopyOperation blobCopyOp;
+public:
+    virtual void startCopy(const char * source, const char *target) override
+    {
+        try
+        {
+            blobClient.reset(new BlobClient(target));
+            blobCopyOp = std::move(blobClient->StartCopyFromUri(source));
```

Comment:
question: AzureBlobClient is very similar to AzureFileClient

To common up the code, could the differences (e.g. the 2 lines above) be placed in derived classes, with the common code in a base?